### ISSUE TITLE
Input not getting input

### ISSUE THREAD

**Issue by elipie** [elipie/BettaOS#1]
The asm input macro is not getting input.... anyone know why?

**Comment by programmeruser2** [comment]
This looks like a complete copy + paste of AmazingMech's tutorial code.
What you need to understand is that there are two types of modes: Real and Protected. There's also Unreal but I won't be talking about it here.
In Real Mode, the BIOS has just loaded you in and you are in Real Mode. This mode is only present on modern x86 processors for backwards compatibility with the 8086/8088.
In this mode, you have access to BIOS interrupts, but you also have some limitations. 
Once you switch to 32-bit real mode, which is usually done by a bootloader such as GRUB, and what invoking QEMU with the 
`-kernel` option does, those restrictions are removed, but you don't have access to BIOS interrupts anymore.
In 32-bit protected mode, you'll have to set up the GDT (only needed if you use a bootloader that you didn't write) and the IDT, after which you can setup IRQs to receive keyboard input.
I'd recommend actually understanding what's happening in the code and learning from https://wiki.osdev.org.

**Comment by elipie** [comment]
Yes, this is just an old project that i was working on. It is a complete copy, and I dont know why i put an issue up. I wanted to learn more about asm/os from this. Thanks!
